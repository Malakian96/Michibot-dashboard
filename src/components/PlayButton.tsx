/* eslint-disable @typescript-eslint/no-explicit-any */
const PlayButton = ({ selectedGuild, selectedChannel, handlePlaySound, disabled }: any) => {
    return (
      <button
        onClick={handlePlaySound}
        disabled={!selectedGuild || !selectedChannel || disabled}
        className={`px-4 py-2 rounded-md ${
          selectedGuild && selectedChannel && !disabled
            ? "bg-discord-blurple text-white hover:bg-opacity-90"
            : "bg-discord-gray text-opacity-50 cursor-not-allowed"
        }`}
      >
        Play Sound
      </button>
    );
  };
  
  export default PlayButton;
  