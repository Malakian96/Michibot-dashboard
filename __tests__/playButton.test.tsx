import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import PlayButton from '../src/components/PlayButton';

describe('PlayButton', () => {
  it('renders correctly when enabled', () => {
    const { getByText } = render(
      <PlayButton
        selectedGuild="1"
        selectedChannel="1"
        handlePlaySound={() => {}}
        disabled={false}
      />,
    );
    expect(getByText('Play Sound')).toBeInTheDocument();
    expect(getByText('Play Sound')).not.toHaveClass('cursor-not-allowed');
  });

  it('renders correctly when disabled', () => {
    const { getByText } = render(
      <PlayButton
        selectedGuild={null}
        selectedChannel={null}
        handlePlaySound={() => {}}
        disabled={true}
      />,
    );
    expect(getByText('Play Sound')).toBeInTheDocument();
    expect(getByText('Play Sound')).toHaveClass('cursor-not-allowed');
  });

  it('calls handlePlaySound when clicked', () => {
    const handlePlaySound = jest.fn();
    const { getByText } = render(
      <PlayButton
        selectedGuild="1"
        selectedChannel="1"
        handlePlaySound={handlePlaySound}
        disabled={false}
      />,
    );

    fireEvent.click(getByText('Play Sound'));
    expect(handlePlaySound).toHaveBeenCalled();
  });

  it('does not call handlePlaySound when disabled', () => {
    const handlePlaySound = jest.fn();
    const { getByText } = render(
      <PlayButton
        selectedGuild={null}
        selectedChannel={null}
        handlePlaySound={handlePlaySound}
        disabled={true}
      />,
    );

    fireEvent.click(getByText('Play Sound'));
    expect(handlePlaySound).not.toHaveBeenCalled();
  });
});

describe('PlayButtonSnapshot', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <PlayButton
        selectedGuild="1"
        selectedChannel="1"
        handlePlaySound={() => {}}
        disabled={false}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
