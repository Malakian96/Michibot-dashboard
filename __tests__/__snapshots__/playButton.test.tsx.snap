// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayButtonSnapshot matches snapshot 1`] = `
<div>
  <button
    class="px-4 py-2 rounded-md bg-discord-blurple text-white hover:bg-opacity-90"
  >
    Play Sound
  </button>
</div>
`;
